src/tensor_ops/max_to/cuda_kernel.rs

```diff
@@ -80,11 +82,15 @@ impl super::MaxReduceKernel<f32> for Cuda {
             BroadcastStridesTo::<Src, Ax>::broadcast_strides(&grad_out.shape, grad_out.strides);
         let out_strides: CudaSlice<usize> = self.dev.take_async(out_strides.into())?;
 
-        let inp_numel = grad_inp.shape.num_elements();
-        let cfg = LaunchConfig::for_num_elems(inp_numel as u32);
+        let physical_numel = grad_inp.data.len();
+        let virtual_numel = grad_inp.shape.num_elements();
+        let elems_per_thread = (virtual_numel / physical_numel) as f32;
+
+        let cfg = LaunchConfig::for_num_elems(physical_numel as u32);
```

Comment:
Do we need to configure shared memory here? There's this field for LaunchConfig that hasn't been used yet anywhere https://docs.rs/cudarc/latest/cudarc/driver/safe/struct.LaunchConfig.html#structfield.shared_mem_bytes

Reply:
Per [nvidia's documentation](https://developer.download.nvidia.com/compute/DevZone/docs/html/C/doc/html/group__CUDA__EXEC_gb8f3dc3031b40da29d5f9a7139e52e15.html), this parameter controls the amount of dynamic shared memory available to each thread block. However, we currently don't have to configure dynamic shared memory because we use statically sized shared memory, and we only need 1024 elements at maximum because this is a limit set by CUDA. Despite this code working, it relies on LaunchConfig's [for_num_elements](https://docs.rs/cudarc/latest/src/cudarc/driver/safe.rs.html#639-647) setting a block size of 1024 to not waste memory.